src/test/java/ca/bc/gov/backendstartapi/endpoint/UserEndpointTest.java

```diff
@@ -1,418 +1,340 @@
 import org.springframework.boot.test.mock.mockito.MockBean;
-import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
-import org.springframework.test.web.reactive.server.WebTestClient;
-import org.springframework.web.bind.support.WebExchangeBindException;
-import reactor.core.publisher.Flux;
-import reactor.core.publisher.Mono;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
 
 @ExtendWith(SpringExtension.class)
-@WebFluxTest(controllers = UserEndpoint.class)
-@Import(UserRepository.class)
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+@TestInstance(Lifecycle.PER_CLASS)
 class UserEndpointTest {
 
   private static final String FIRST_NAME = "Ricardo";
   private static final String LAST_NAME = "Campos";
   private static final UserDto USERDTO = new UserDto(FIRST_NAME, LAST_NAME);
 
-  @Autowired private WebTestClient webTestClient;
+  private MockMvc mockMvc;
 
   @MockBean UserRepository userRepository;
 
+  @Autowired private WebApplicationContext webApplicationContext;
+
+  @BeforeEach
+  public void setup() {
+    this.mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+  }
+
+  private String getUserDtoString(UserDto userDto) throws Exception {
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.configure(SerializationFeature.WRAP_ROOT_VALUE, false);
+    ObjectWriter ow = mapper.writer().withDefaultPrettyPrinter();
+    return ow.writeValueAsString(userDto);
+  }
+
   @Test
   @DisplayName("Create user with success")
-  void createSuccess() {
-    Mockito.when(userRepository.save(USERDTO)).thenReturn(Mono.just(USERDTO));
-
-    webTestClient
-        .post()
-        .uri("/users")
-        .contentType(MediaType.APPLICATION_JSON)
-        .accept(MediaType.APPLICATION_JSON)
-        .body(Mono.just(USERDTO), UserDto.class)
-        .exchange()
-        .expectStatus()
-        .isOk()
-        .expectHeader()
-        .contentType(MediaType.APPLICATION_JSON)
-        .expectBody()
-        .jsonPath("$.firstName")
-        .isEqualTo(FIRST_NAME)
-        .jsonPath("$.lastName")
-        .isEqualTo(LAST_NAME);
+  void createSuccess() throws Exception {
+    Mockito.when(userRepository.save(USERDTO)).thenReturn(USERDTO);
+
+    mockMvc
+        .perform(
+            post("/users")
+                .header("Content-Type", "application/json")
+                .accept(MediaType.APPLICATION_JSON)
+                .content(getUserDtoString(USERDTO)))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.firstName").value(FIRST_NAME))
+        .andExpect(jsonPath("$.lastName").value(LAST_NAME))
+        .andReturn();
   }
```

Comment:
Actually, we're testing the endpoint classes, return codes, headers and returning content. The mock part is only to avoid the repository class interaction. But since this is an integration test, and the repository class is just an in-memory handler, we can keep an use the real implementation. I'll fix it.